File: app/public_html/script/functions/samples/29.c
loop(){
    if (getCloseEnemy()){
        if (isRunning() || getDistToTarget() < 2)
            teleport(0,0);
        else
            fireball(getTargetX(), getTargetY());
    }
    else if (!getHit()){
        turn(5);
        stepForward();
    }
    else
        turnToLastHit();
}

File: app/public_html/script/functions/samples/29.py
def loop():
    if getCloseEnemy():
        if isRunning() or getDistToTarget() < 2:
            teleport(0,0)
        else:
            fireball(getTargetX(), getTargetY())
     
    elif not getHit():
        turn(5)
        stepForward()
     
    else:
        turnToLastHit()
 